### PR TITLE
Switching to conda-sphinx-theme plus reorganizing the site a bit

### PR DESCRIPTION
**Pull request by travishathaway**
### Description

Fixes: #367 

This pull request also re-organizes the website. I did this to accommodate the new menu structure. In the top bar, links to other conda projects will be shown alongside `conda-libmamba-solver`. On the sidebar, I have chosen to organize the docs site into two sections: "User guide" and "Developer guide". The front page was also changed a little bit.

I know these are big changes so please give me feedback. I would be happy to make changes.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

File: news/370-use-conda-sphinx-theme [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+* <news item>
+
+### Deprecations
+
+* <news item>
+
+### Docs
+
+* Use new conda-sphinx-theme for documentation site
+* Reorganize the layout of the documentation site
```
Comment (jaimergp) [inline news/370-use-conda-sphinx-theme:16]:
```suggestion
* Use new conda-sphinx-theme for documentation site. (#367 via #370)
* Reorganize the layout of the documentation site. (#370)
```

**Review by jaimergp — APPROVED** [review]
Thanks! Looks awesome. You even took care of the redirects! <3

**Comment by jaimergp** [comment]
cc @jezdez who originally put the site together, in case there are further comments. Otherwise I'd merge :)

**Comment by jezdez** [comment]
lemme take a look when I'm back at a computer before merge

**Comment by jaimergp** [comment]
Oh, I see. Cool! Without it it's a bit less brand-y (only the accent color is conda green) but otherwise it looks good in terms of contrast.

**Comment by travishathaway** [comment]
@jaimergp,

I just released a new version of the conda-sphinx-theme that keeps the link colors green and stays, "on brand" 😄.

File: docs/_templates/navbar_center.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<ul id="navbar-main-elements" class="navbar-nav">
+    <li class="nav-item current active">
+        <a class="reference internal nav-link" href="{{ pathto("index") }}">{{ project }}</a>
+    </li>
+    <li class="nav-item {% if project == "conda" %}current active{% endif %}">
+        <a class="reference internal nav-link" href="https://docs.conda.io/en/latest/conda.html">Conda</a>
```
Comment (jezdez) [inline docs/_templates/navbar_center.html:6]:
```suggestion
        <a class="reference internal nav-link" href="https://docs.conda.io/projects/conda/">Conda</a>
```

File: docs/_templates/navbar_center.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    </li>
+    <li class="nav-item {% if project == "conda-build" %}current active{% endif %}">
+        <a class="reference internal nav-link" href="https://docs.conda.io/en/latest/conda-build.html">Conda-build</a>
```
Comment (jezdez) [inline docs/_templates/navbar_center.html:9]:
```suggestion
        <a class="reference internal nav-link" href="https://docs.conda.io/projects/conda-build/">Conda-build</a>
```

File: docs/_templates/navbar_center.html [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    </li>
+    <li class="nav-item {% if project == "miniconda" %}current active{% endif %}">
+        <a class="reference internal nav-link" href="https://docs.conda.io/en/latest/miniconda.html">Miniconda</a>
```
Comment (jezdez) [inline docs/_templates/navbar_center.html:12]:
```suggestion
        <a class="reference internal nav-link" href="https://docs.conda.io/projects/miniconda/">Miniconda</a>
```

**Review by jezdez — APPROVED** [review]
The links to the other docs sites shouldn't go through the conda-docs, and instead link to the subproject URLs. Less redirects that way.

**Comment by travishathaway** [comment]
@jaimergp and @jezdez, thanks for the review. I do not have the ability to merge this, so one of you will have to do it.

File: docs/conf.py [hunk 5]
```diff
@@ -84,3 +100,15 @@
...
+
+redirects = {
+    "getting-started": "user-guide",
+    "faq": "user-guide/faq",
+    "configuration": "user-guide/configuration",
+    "libmamba-vs-classic": "user-guide/libmamba-vs-classic",
+    "more-resources": "user-guide/more-resources",
+    "performance": "user-guide/performance",
+    "subcommands": "user-guide/subcommands",
+}
```
Comment (jezdez) [inline docs/conf.py:114]: Something was wrong with this section: https://github.com/conda/conda-libmamba-solver/issues/374

**Pull request merged**